src/test/kotlin/stringAddCalculatorTest/StringAddCalculatorTest.kt

```diff
@@ -0,0 +1,54 @@
+package stringAddCalculatorTest
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import org.junit.jupiter.params.provider.NullAndEmptySource
+import stringAddCalculator.StringAddCalculator
+
+class StringAddCalculatorTest {
+    private lateinit var calculator: StringAddCalculator
+
+    @BeforeEach
+    fun setUp() {
+        calculator = StringAddCalculator()
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    fun `입력한 문자열이 빈 문자열 또는 null이라면 0을 반환`(text: String?) {
+        assertThat(calculator.calculate(text)).isZero()
+    }
+
+    @ParameterizedTest
+    @MethodSource("generateInputAndExpected")
+    fun `입력한 문자열의 합 계산`(text: String, expected: Int) {
+        assertEquals(expected, calculator.calculate(text))
+    }
+
+    companion object {
+        @JvmStatic
+        private fun generateInputAndExpected(): List<Arguments?> {
+            return listOf(
+                Arguments.of(
+                    "1,2,3", 6
+                ),
+                Arguments.of(
+                    "1,2:3", 6
+                ),
+                Arguments.of(
+                    "1:2:3", 6
+                ),
+                Arguments.of(
+                    "//!\n1!2!3", 6
+                ),
+                Arguments.of(
+                    "//!\n1:2!3,4", 10
+                ),
+            )
+        }
+    }
```

Comment:
p1; 테스트 코드가 실패해서 확인해 보니, \n 제어 문자를 그대로 입력해서 줄바꿈되어 들어가네요.
![image](https://github.com/next-step/kotlin-lotto/assets/77874587/7580a181-903a-4ac3-b8cb-f25385184321)

```suggestion
    companion object {
        @JvmStatic
        private fun generateInputAndExpected(): List<Arguments> = listOf(
            Arguments.of(
                """//!\n1!2!3""", 6
            ),
            Arguments.of(
                """//!\n1:2!3,4""", 10
            ),
        )
    }
```

코틀린의 삼중따옴표를 사용해 보면 좋을 거 같아요. 👍
![image](https://github.com/next-step/kotlin-lotto/assets/77874587/02f69c01-3719-4526-86c7-86520d69024e)

Reply:
어라.. 제가 돌리면 에러가 발생하지 않는데 어디서 잘못된 것일까요..??

그와 별개로 삼중따옴표에 대해서 확인차 여쭙고 싶은 것이 있습니다!
nextstep 강의 자료에서는 삼중따옴표를 사용하지 않고 계셔서 제어 문자가 구분자로 작동하는 것이 요구사항으로 이해했었습니다.
그래서 의도적으로 삼중따옴표를 사용하지 않았는데요, 혹시 "\n"이라는 문자를 구분자로 사용하도록 구현하면 될까요?

<img width="547" alt="스크린샷 2023-11-04 오후 4 40 52" src="https://github.com/next-step/kotlin-lotto/assets/47503412/6dd097a8-b1cc-4978-8630-559925fe2e7a">

<img width="504" alt="스크린샷 2023-11-04 오후 4 37 58" src="https://github.com/next-step/kotlin-lotto/assets/47503412/0778db54-0cab-4deb-9843-46e4123376c5">

<img width="567" alt="스크린샷 2023-11-04 오후 4 35 05" src="https://github.com/next-step/kotlin-lotto/assets/47503412/5df7cc30-dcdc-4511-a925-3bced5264f92">

Reply:
@10000001a 
`예를 들어 “//;\n1;2;3”과 같이 값을 입력할 경우 커스텀 구분자는 세미콜론(;)이며, 결과 값은 6이 반환되어야 한다.`
이라는 기능 요구 사항이 있어, 콘솔 입력 기준으로 판단했는데 테스트 코드에 그렇게 있다면 공백으로 감안하고 보겠습니다. 😊
다른 코멘트만 확인해 주시면 될 거 같아요.

Reply:
답변 감사합니다! 나머지 부분 마무리해서 다시 리뷰 요청드리겠습니다🙇‍♂️🙇‍♂️